colossalai/accelerator/api.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+from typing import Union
+from .base_accelerator import BaseAccelerator
+from .cuda_accelerator import CudaAccelerator
+from .npu_accelerator import NpuAccelerator
+
+__all__ = ['set_accelerator', 'get_accelerator']
+
+
+_ACCELERATOR = None
+
+_ACCELERATOR_MAPPING = {
+    "cuda": CudaAccelerator,
+    "npu": NpuAccelerator,
+}
+
+_DEFAULT_ACCELERATOR_TYPE = CudaAccelerator
+
+def set_accelerator(accelerator: Union[str, BaseAccelerator]) -> None:
```

Comment:
Do we have `is_available()` or similar api?

Reply:
Yes, plans to update `colossalai.utils.device` in a separate PR.

Reply:
whether an accelerator can be truely used. E.g. before/when setting accelerator, we have to know whether it can be truely used.

Reply:
I guess we can only do this by keeping/removing an accelerator from `_ACCELERATOR_MAPPING`

Reply:
Or when auto select accelerator, we have to detect it. Most machines only have one type of accelerator. We may have a priority list and detect the availability

Reply:
It can be difficult for this module to take the responsibility to check whether it works for all other features. Instead, other features should keep a list of supported accelerators on their own.

Reply:
> Or when auto select accelerator, we have to detect it. Most machines only have one type of accelerator. We may have a priority list and detect the availability

Maybe, I can do this?

```diff
def set_accelerator(accelerator: Union[str, BaseAccelerator]) -> None:
    """
    Set the global accelerator for the current process.

    Args:
        accelerator (Union[str, BaseAccelerator]): the type of accelerator to which the current device belongs.
    """

    global _ACCELERATOR

    if isinstance(accelerator, str):
        _ACCELERATOR = _ACCELERATOR_MAPPING[accelerator]()
    elif isinstance(accelerator, BaseAccelerator):
        _ACCELERATOR = accelerator
    else:
        raise TypeError("accelerator must be either a string or an instance of BaseAccelerator")
+   assert _ACCELERATOR.is_available()
```

Reply:
OK. But in this way, users have to choose accelerator on their own. Can users transfer their code from cuda to npu without any cost? Now, we choose the device when calling `launch()` and users don't need to do anything related to device.

Reply:
Or we have one more api like this?

```python
def auto_set_accelerator():
     # check if any accelerator is available
     # if available, set it to the global var
```

Reply:
LGTM

Reply:
Added.